src/graph.rs

```diff
@@ -27,27 +29,55 @@ impl GitGraph {
             repository,
             commits,
         };
-        graph.assign_branches(indices)?;
+        graph.assign_branches(indices, settings)?;
 
         Ok(graph)
     }
-    fn assign_branches(&mut self, indices: HashMap<Oid, usize>) -> Result<(), Error> {
-        let branches = self.repository.branches(None)?;
-        for branch in branches {
-            let (branch, branch_type) = branch?;
-            if branch_type == BranchType::Local {
-                let reference = branch.get();
-                if let Some(name) = reference.name() {
-                    if let Some(oid) = reference.target() {
-                        let idx = indices[&oid];
-                        self.commits[idx].branches.push(name[11..].to_string());
-                    }
+    fn assign_branches(
+        &mut self,
+        indices: HashMap<Oid, usize>,
+        settings: &Settings,
+    ) -> Result<(), Error> {
+        let branches_ordered = Self::branches_persistence_order(&self.repository, settings)?;
+
+        for branch in branches_ordered {
+            let reference = branch.get();
+            if let Some(name) = reference.name() {
+                if let Some(oid) = reference.target() {
+                    let idx = indices[&oid];
+                    self.commits[idx].branches.push(name[11..].to_string());
                 }
             }
         }
 
         Ok(())
     }
+
+    fn branches_persistence_order<'repo>(
+        repository: &'repo Repository,
+        settings: &Settings,
+    ) -> Result<Vec<Branch<'repo>>, Error> {
+        let branches = repository
+            .branches(Some(BranchType::Local))?
+            .filter_map(|bt| bt.ok().map(|bt| bt.0))
+            .map(|branch| {
+                (
+                    settings
+                        .branch_persistance
+                        .iter()
+                        .position(|b| {
+                            branch.get().name().unwrap_or("refs/heads/-")[11..].starts_with(b)
+                        })
+                        .unwrap_or(1000),
```

Comment:
Instead of this ad-hoc value, couldn't `branch_persistance.len()` be used?

Reply:
Don't know why I didn't have that idea... 😉